GNUmakefile

```diff
@@ -86,9 +86,9 @@ cleanest: cleaner
 requirements: export CUSTOM_COMPILE_COMMAND='make requirements'
 requirements:
 	pip install --upgrade pip setuptools pip-tools
-	cd requirements && pip-compile --upgrade --resolver=backtracking requirements.in
-	cd requirements && pip-compile --upgrade --resolver=backtracking requirements-dev.in
-	cd requirements && pip-compile --upgrade --resolver=backtracking requirements-typing.in
+	pip-compile --upgrade --resolver=backtracking requirements/requirements.in -o requirements/requirements.txt
```

Comment:
Note: the reason this breaks is because the use of `.ONESHELL:`. That feature doesn't exist in the version of make that ships with macs, resulting in diverging behaviors.

Once we have a [more complete docker dev env](https://github.com/Aiven-Open/karapace/issues/585), I'll argue for removing much of the makefile targets that downloads executables and sets up venvs and so on.